kura/test/org.eclipse.kura.nm.test/src/test/java/org/eclipse/kura/nm/NMDeviceStateTest.java

```diff
@@ -182,9 +183,97 @@ public void shouldReturnConnectionTrueFailed() {
+        thenStateUIntShouldBeEqualTo(new UInt32(10));
+    }
+
+    @Test
+    public void conversionWorksForStateUnavailableUINT() {
+        whenNMDeviceStateStateIsPassed(NMDeviceState.NM_DEVICE_STATE_UNAVAILABLE);
+        thenStateUIntShouldBeEqualTo(new UInt32(20));
+    }
+
+    @Test
+    public void conversionWorksForStateDisconnectedUINT() {
+        whenNMDeviceStateStateIsPassed(NMDeviceState.NM_DEVICE_STATE_DISCONNECTED);
+        thenStateUIntShouldBeEqualTo(new UInt32(30));
+    }
 
-    public void whenInt32StateIsPassed(UInt32 type) {
-        this.state = NMDeviceState.fromUInt32(type);
+    @Test
+    public void conversionWorksForStatePrepareUINT() {
+        whenNMDeviceStateStateIsPassed(NMDeviceState.NM_DEVICE_STATE_PREPARE);
+        thenStateUIntShouldBeEqualTo(new UInt32(40));
+    }
+
+    @Test
+    public void conversionWorksForStateConfigUINT() {
+        whenNMDeviceStateStateIsPassed(NMDeviceState.NM_DEVICE_STATE_CONFIG);
+        thenStateUIntShouldBeEqualTo(new UInt32(50));
+    }
+
+    @Test
+    public void conversionWorksForStateNeedAuthUINT() {
+        whenNMDeviceStateStateIsPassed(NMDeviceState.NM_DEVICE_STATE_NEED_AUTH);
+        thenStateUIntShouldBeEqualTo(new UInt32(60));
+    }
+
+    @Test
+    public void conversionWorksForStateIpConfigUINT() {
+        whenNMDeviceStateStateIsPassed(NMDeviceState.NM_DEVICE_STATE_IP_CONFIG);
+        thenStateUIntShouldBeEqualTo(new UInt32(70));
+    }
+
+    @Test
+    public void conversionWorksForStateIpCheckUINT() {
+        whenNMDeviceStateStateIsPassed(NMDeviceState.NM_DEVICE_STATE_IP_CHECK);
+        thenStateUIntShouldBeEqualTo(new UInt32(80));
+    }
+
+    @Test
+    public void conversionWorksForStateSecondariesUINT() {
+        whenNMDeviceStateStateIsPassed(NMDeviceState.NM_DEVICE_STATE_SECONDARIES);
+        thenStateUIntShouldBeEqualTo(new UInt32(90));
+    }
+
+    @Test
+    public void conversionWorksForStateActivatedUINT() {
+        whenNMDeviceStateStateIsPassed(NMDeviceState.NM_DEVICE_STATE_ACTIVATED);
+        thenStateUIntShouldBeEqualTo(new UInt32(100));
+    }
+
+    @Test
+    public void conversionWorksForStateDeactivatingUINT() {
+        whenNMDeviceStateStateIsPassed(NMDeviceState.NM_DEVICE_STATE_DEACTIVATING);
+        thenStateUIntShouldBeEqualTo(new UInt32(110));
+    }
+
+    @Test
+    public void conversionWorksForStateFailedUINT() {
+        whenNMDeviceStateStateIsPassed(NMDeviceState.NM_DEVICE_STATE_FAILED);
+        thenStateUIntShouldBeEqualTo(new UInt32(120));
+    }
+
+    @Test
+    public void conversionWorksForStateUnknownDefaultUINT() {
+        whenNMDeviceStateStateIsPassed(NMDeviceState.NM_DEVICE_STATE_UNKNOWN);
+        thenStateUIntShouldBeEqualTo(new UInt32(0));
+    }
+
+    public void whenInt32StateIsPassed(UInt32 state) {
+        this.state = NMDeviceState.fromUInt32(state);
+    }
+    
+    public void whenNMDeviceStateStateIsPassed(NMDeviceState state) {
```

Comment:
Some mishap with the rename method tool:

```suggestion
    public void whenNMDeviceStateIsPassed(NMDeviceState state) {
```